modules/core/src/main/java/org/apache/ignite/internal/processors/rest/GridRestProcessor.java

```diff
@@ -150,7 +150,11 @@
         }
 
         @Override public IgniteInternalFuture<GridRestResponse> handleAsync(GridRestRequest req) {
-            return handleAsync0(req);
+            IgniteInternalFuture<GridRestResponse> res = handleAsync0(req);
+
+            res.listen(fut -> logRestRequestResult(req, fut));
```

Comment:
Can we use closure here?

Reply:
No, we can't. We need GridRestRequest to be logged and IgniteInternalFuture result doesn't provide it.

Reply:
```
res.listen(fuct -> {
});
```

Why this wouldn't work?